test/release/examples/patterns/readcsv.chpl

```diff
@@ -0,0 +1,286 @@
+    for colIdx in 0..colNames.size-2 {
+      writer.write(colNames[colIdx],",");
+    }
+    // The last column name should not be followed by comma.
+    writer.writeln(colNames[colNames.size-1]);
+
+    // Then write to the file each row of data.
+    for row in dataRows {
+      for colIdx in 0..colNames.size-2 {
+        writer.write(row[colNames[colIdx]],",");
+      }
+      // The last column value should not be followed by comma.
+      writer.writeln(row[colNames[colNames.size-1]]);
+    }
+    writer.close();
+
+    // Compare the file generated from the list of maps with the original input csv file.
+    checkThatFilesMatch(inFileName,approach1FileName);
+  }
+}
+
+// Approach 2: chapeltastic -- create an associative array with the key being the 
+//             column name and the value being an array of strings
+{
+  writeln();
+  writeln();
+  writeln("Approach 2: associative array keyed of column name with array of vals per column"); 
+
+  // Create another reader of the input csv file
+  var reader = f.reader();
+
+  // Read the first line to get the column names.
+  // Note: this portion is the same as in Approach 1
+  var line : string;
+
+  if (!reader.readline(line)) then
+    writeln("ERROR: ", inFileName, " appears to be empty");
+
+  var colNames : list(string);
+  colNames = createListOfColNames(line);
+
+  if debug {
+    writeln("\ncolNames: ", colNames);
+  }
+
+  // Create a list of lines from the file, so we know how many lines of data there are.
+  // Then create an associative array with an entry per column of data, with the
+  // associative key being the column name.  The value will be an array of strings
+  // with the column's value for each row in the data set.
+  var dataRows : list(string);
+
+  // Reading all of the lines of the file into a list.
+  while (reader.readline(line)) {
+    dataRows.append(line);
+  }
+
+  // Declaring an associative array, where the value type is a 1D array large enough
+  // to store one column value per row of data.
+  var valDomain = {0..dataRows.size-1}; // Domain indicating how many values per column.
+  var colNameDomain: domain(string);    // An associative domain of strings.
+  for colName in colNames do            // Put each column name into the column name domain.
+    colNameDomain += colName;
+  // colData is an associative array indexed by the column name domain with array values.
+  // Each array value will have number of rows string elements.
+  var colData: [colNameDomain] [valDomain] string;
+
+  // Processing each row to put the column values into the associative array.
+  for rowIdx in 0..dataRows.size-1 {
+    // Pull the line out from the list of dataRows.
+    line = dataRows[rowIdx];
+    var start = 0;      // index into the line string
+    var colIdx = 0;     // the first column is indexed as zero
+    // Process each line by finding the next field value from the start index to the next comma.
+    while (start<line.size && colIdx<colNames.size) {
```

Comment:
Similar change to `for` as above?

Reply:
Right, I changed this one to using the field iterator you sent in slack.  Copied here for reference:

```
    for (colName, field) in zip(colNames, fieldsInRow(line)) {
      colData[colName][rowIdx] = field;
    }

iter fieldsInRow(line) {
  var start = 0;
  while start<line.size {
    var (nextVal, nextCommaIdx) = nextField(line, start);
    start = nextCommaIdx + 1; // The next field will start after the comma.
    yield nextVal;
  }
}

```